Christmas dev

anastasiyamnv opened the pull request.

anastasiyamnv committed:
Add files via upload

anastasiyamnv committed:
Create text.txt

anastasiyamnv committed:
Add files via upload

anastasiyamnv committed:
Delete images/text.txt

anastasiyamnv committed:
Add files via upload

anastasiyamnv committed:
Add files via upload

anastasiyamnv committed:
fix: fixed footer cards position

anastasiyamnv committed:
feat: add 'The Gifts' page, burger menu, 'Back to Top' button, and card filtering functionality

Implemented the main layout for 'The Gifts' page.
Added a responsive burger menu for navigation.
Created a 'Back to Top' button that scrolls to the top of the page.
Added functionality to filter cards by categories such as 'For Work', 'For Health', and 'For Harmony'.

K2-m5 commented on gifts.css:
```diff
@@ -0,0 +1,98 @@
+
+.best-gifts-card.for-work .best-gifts-card__image {
+    background-image: url('images/gift-for-work.png');
+    background-repeat: no-repeat;
```
для класса .best-gifts-card__image повторяются свойства бекграунд рипет, позиция и размер, предлагаю их вынести в общий класс

K2-m5 commented on gifts.css:
```diff
@@ -0,0 +1,98 @@
+    background-size: 100% 230px;
+}
+
+.best-gifts-card__image {
```
background-repeat: no-repeat;
    background-position: top;
    background-size: 100% 230px;
вот сюда их добавить

K2-m5 commented on gifts.html:
```diff
@@ -0,0 +1,190 @@
+
+                <div class="best-gifts-section__cards">
+                    <div class="best-gifts-card for-work">
+                        <div class="best-gifts-card__image" style="background-image: url('images/gift-for-work.png');"></div>
```
наверное не стоит дублировать background-image, ты добавила их в css

K2-m5 commented on gifts.html:
```diff
@@ -0,0 +1,190 @@
+                    </div>
+                    
+                    <div class="best-gifts-card for-health">
+                        <div class="best-gifts-card__image" style="background-image: url('images/gift-for-health.png');"></div>
```
то же самое для всех карточек

K2-m5 commented on images/X.png:
поправь название файла на маленькую букву

K2-m5 commented on gifts.html:
```diff
@@ -0,0 +1,190 @@
+                        <img src="images/instagram.png" alt="Instagram">
+                    </a>
+                    <a href="https://x.com" class="footer__social-media-link">
+                        <img src="images/x.png" alt="X">
```
у тебя файл с картинкой с большой буквы, поэтому на странице картинка не подтягивается

K2-m5 commented on index.html:
```diff
@@ -0,0 +1,180 @@
+<body>
+    <header class="header">
+        <div class="header__line">
+            <a href="#" class="header__logo">
```
тут нужно добавить редирект на главную страницу то есть домой на индекс файл

K2-m5 commented on gifts.html:
```diff
@@ -0,0 +1,190 @@
+<body>
+    <header class="header">
+        <div class="header__line">
+            <a href="#" class="header__logo">
```
добавить редирект на главную

K2-m5 commented on index.html:
```diff
@@ -0,0 +1,180 @@
+            </div>
+        </footer> 
+    </div> 
+<script src="script.js"></script>  
```
поправь отступы пожалуйста

K2-m5 commented on gifts.html:
```diff
@@ -0,0 +1,190 @@
+        <div class="arrow-right"></div>
+    </button>
+    
+<script src="script.js"></script>
```
поправь отступы пожалуйста

K2-m5 commented on styles.css:
```diff
@@ -0,0 +1,775 @@
+.footer {
+    max-width: 1440px; 
+    width: 100%;
+    margin: 60px auto;  
```
центрирование контента ты уже сделала спомощью flex, скорее всего марджин auto тебе не нужен

K2-m5 commented on styles.css:
```diff
@@ -0,0 +1,775 @@
+    align-items: center;
+}
+
+.cta-section__button:hover {
```
не нашел тега с таким классом

K2-m5 commented on styles.css:
```diff
@@ -0,0 +1,775 @@
+
+.best-gifts-card:nth-child(1) {
+    background-image: url('images/gift-for-work.png');
+    background-repeat: no-repeat;   
```
такой коммент я уже писал выше, повторяющиеся свойства стилей лучше вынести в общий стили для компонента. Не очень нравиться обращение через чайлдов, сделай доп классами как на странице gift

K2-m5 commented on styles.css:
```diff
@@ -0,0 +1,775 @@
+    background-size: 100% 230px;   
+}
+
+.best-gifts-card:nth-child(2) {
```
то же

K2-m5 commented on script.js:
```diff
@@ -0,0 +1,41 @@
+    }
+};
+
+document.getElementById('back-to-top').onclick = function () {
```
давай тут использовать addEventListner

K2-m5 commented on script.js:
```diff
@@ -0,0 +1,41 @@
+        const cards = document.querySelectorAll('.best-gifts-card');
+        
+        cards.forEach(card => {
+            if (category === 'all') {
```
давай избавимся от else if и сделаем тут такое условие 
if (category === 'all' || card.classList.contains(`for-${category}`)) .....

anastasiyamnv committed:
fix: fixed Scroll-to-Top button logic, improved burger menu functionality, and refactored styles by extracting common properties and removing child selectors, changed image filename to lowercase

feat: added slider logic, countdown timer logic, implemented redirect to the homepage

anastasiyamnv committed:
feat: add dynamic loading of gift cards

anastasiyamnv committed:
feat: Add modal for the selected product on both pages

anastasiyamnv committed:
refactor: updated block comments and optimized the code slightly

anastasiyamnv committed:
refactor

anastasiyamnv committed:
fix: corrected indentation and formatting issues

anastasiyamnv committed:
Moved common animation styles to global for the slider container, updated variable declarations.

anastasiyamnv committed:
Merge pull request #4 from anastasiyamnv/christmas-dev-second

Corrected errors and added functionality
